cmux-tui/scripts/check-spec-inventory.py

```
@@ -0,0 +1,436 @@
+    print(f"spec inventory error: {message}", file=sys.stderr)
+    raise SystemExit(1)
+
+
+def unique(values: list[str], label: str) -> set[str]:
+    duplicates = sorted({value for value in values if values.count(value) > 1})
+    if duplicates:
+        fail(f"duplicate {label}: {', '.join(duplicates)}")
+    return set(values)
+
+
+def validate_schema(value: object, schema: dict[str, object], path: str = "$") -> None:
```

Ruff (0.15.21)

[warning] Too many branches (21 > 12)

(PLR0912)